packages/enzyme-adapter-utils/src/createMountWrapper.jsx

```diff
@@ -32,6 +32,29 @@ const makeValidElementType = (adapter) => {
   return validElementType;
 };
 
+/**
+ * This function flattens a list of react elements into a tree:
+ *
+ * For example:
+ *
+ *  In: [<Foo hello="world" />, <Bar name="Jane Doe" />, <Baz value={27} />]
+ *  Out: (
+ *    <Foo hello="world" />
+ *      <Bar name="Jane Doe">
+ *        <Baz value={27} />
+ *      </Bar>
+ *    </Foo>
+ *  )
+ */
+const reduceElements = (elements) => {
+  const reversed = [...elements];
+  reversed.reverse();
+
+  return reversed.reduce((childElement, parentElement) => (
+    React.cloneElement(parentElement, null, childElement)
```

Comment:
cloning elements is very slow and should be avoided. why is it needed here?

Reply:
The goal is to convert the provider elements the user passes into a tree:

So this:
```jsx
mount(<MyComponent />, {
  providers: [
    <ContextA.Provider value="A" />,
    <ContextB.Provider value="B" />,
    <ContextC.Provider value="C" />,
  ],
});
```

Is rendered as:
```jsx
<ContextA.Provider value="A">
  <ContextB.Provider value="B">
    <ContextC.Provider value="C">
      <MyComponent />
    </ContextC.Provider>
  </ContextB.Provider>
</ContextA.Provider>
```

And I figured `cloneElement()` was the most idiomatic way to do this. I'm also working under the assumption that _mutating_ the providers the user passes in would be a bad idea. Would a more manual
```jsx
// not sure if this will actually work
{
  ...parentElement,
  props: {
    ...parentElement.props,
    children: childElement,
  },
}
```

be preferable, or is it the copying itself that is too slow to be acceptable?